package/MDAnalysis/analysis/base.py

```diff
@@ -125,14 +125,105 @@
 import inspect
 import logging
 import itertools
+import warnings
 
 import numpy as np
 from MDAnalysis import coordinates
 from MDAnalysis.core.groups import AtomGroup
 from MDAnalysis.lib.log import ProgressBar
+from typing import Callable, Iterable
+from importlib import import_module
+from functools import partial
 
 logger = logging.getLogger(__name__)
 
+class ParallelExecutor(object):
+    """Class that executes parallel computations for AnalysisBase (but potentially for other purposes)
+    using one of the available backends"""
+
+    @classmethod
+    @property
+    def possible_backends(cls):
+        return ("local", "dask", "multiprocessing", "dask.distributed")
+
+    def __init__(self, backend: str, n_workers: int, client: object = None):
+        # validate 'client' argument
+        if client is not None and backend is not None:
+            raise ValueError(
```

Comment:
Please add tests for all these errors

Reply:
Agree. It is very tedious but important. And many Codecov annotations make reviewing more difficult (interestingly, they are not rendered on the mobile app).